Feature/mypage

daindaind opened the pull request:
중간 확인 부탁드립니다.. 

avatar 누르면 마이페이지 뜨는 건 아직 구현 안 되어있어요](https://github.com/SMUING-D/FE-Next/pull/9)

daindaind committed:
feat(userinfoedit usermenu getuserinfo mypage handler data): mypage 구현

daindaind committed:
feat: userInfoModal 생성

daindaind committed:
feat: userInfoEditModal, userInfo schema, editUserInfo 수정

daindaind committed:
feat: passwordEditModal 구현, middleware mypage

daindaind committed:
feat: modal component, mypage darkmode

daindaind committed:
feat: console.log errors delete

dydals3440 commented:
- [ ] 학교에서 무슨 활동 했는지, 정리할 수 있는 칸 필요! UMC 4기 -> UMC 5기 파트장 등등 이런느낌으로!
- [x] https://www.inflearn.com/users/213477 인프런 사이트는 이런식으로, 다양한 유저들의 마이페이지를 조회하기 위해 유저 아이디로 받고 있습니당!! 우리만, 우리의 마이페이지를 보는 것이 아니라, 이제 다른 사람들의 댓글이미지나, 글쓴이 이미지를 클릭하면 그 사람의 마이페이지를 볼 수 있으므로 경로를 http://localhost:3000/mypage가 아니라 http://localhost:3000/mypage/userId 로 변경하거나, 아니면 http://localhost:3000/user/userId로 변경하는게 좋아보여요!!
- [x] 이미지 보낼떄, 만약에 압축파일이나 다른 파일 보내면 그것도 자동으로 첨부되고 보내지는거같은데, 이부분도 살짝 거를 수 있으면 거르면 좋을 것 같습니당!!

daindaind committed:
feat: user/:userId 로 경로 변경, middleware, next-auth 수정

daindaind committed:
feat: commentView, Mypage userId 확인 후 표시하는 것으로 변경

authorize 반환 타입 에러는 이후에 어떻게 될지 몰라서 그냥 놔뒀구요.. session data type을 좀 변경했습니다. NextAuth 써놓은 것은 그냥 문법 때문에
임시방편으로 해 둔 것입니다.

daindaind committed:
fix: file validation 안되는 오류 수정했습니다

daindaind committed:
fix(userinfo): 에러 메시지 한글로 표기

dydals3440 commented:
### Todo
<img width="1205" alt="스크린샷 2024-01-10 오전 4 02 37" src="https://github.com/SMUING-D/FE-Next/assets/119042360/7d8066c8-b3dc-44fd-823f-97318187638f">
- [ ] 데이터 통신을 못했을 경우 분기처리. -> 사용자에게 유저 정보를 못찾았다는 문구 페이지 띄워주면 좋을 것 같아요!! 제가 만든 ErrorState 컴포넌트 참고!
<img width="1205" alt="스크린샷 2024-01-10 오전 4 02 37" src="https://github.com/SMUING-D/FE-Next/assets/119042360/ac9d91d9-d189-470f-a4c0-f90697543dec">
- [ ] 일단은, 정상적인 이미지 넣고 -> 앱 어플리케이션 (깃허브 앱)넣은후 -> 모달이 멈췄습니다!(이미지 초기화도 안먹고 파일 재업로드도 안되는 것 같습니다~!! 재확인 부탁드려용) 만약에 안되면 같이 고민해보죠
- [ ] 마이페이지에 졸업 유무 학생인지, 직장인인지 이런거 좀 적어주면 좋을 것 같다는 것이 저의 의견!! 혹시 어떻게 생각하는지에 대한 여부도 알고싶습니다!!

고생 많으셨습니당~😄

daindaind committed:
fix: data fetch error errorpage, 이미지 초기화

daindaind committed:
fix: 이미지 선택 후 뒤로가면 파일명 사라지는 error fix

FileInput, UserInfoEditModal

daindaind committed:
fix: 에러 처리

daindaind committed:
fix: input file accept 수정
